sdk/src/nonce_account.rs

```diff
@@ -20,6 +20,9 @@ pub fn create_account(lamports: u64) -> RefCell<Account> {
 }
 
 pub fn verify_nonce_account(acc: &Account, hash: &Hash) -> bool {
+    if acc.owner != crate::system_program::id() {
+        return false;
+    }
```

Comment:
just making sure: is it ok to omit gating here? I think this creates some time window when updating where specially crafted transactions could result in differing TransactionStatus.

I know there should be little possibility for the differing behavior under normal validator operation. So, this could be safe?

Reply:
Put differentially, not gating here could be justified because we're kind of fixing security issues?

Reply:
I think this one is OK since if we hit it, halting the network is probably the better outcome

Reply:
ah i see.